src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/withToolbar.js

```diff
@@ -11,23 +11,29 @@ export default function withToolbar(
     toolbar: () => ToolbarConfig,
     toolbarStoreKey: string = DEFAULT_STORE_KEY
 ) {
-    const WithToolbarComponent = class extends React.Component<*> {
-        disposer: Function;
-
-        setToolbarConfig = (component: React$Element<*>) => {
-            this.disposer = autorun(() => {
-                if (component) {
-                    toolbarStorePool.setToolbarConfig(toolbarStoreKey, toolbar.call(component));
-                }
+    const WithToolbarComponent = class extends Component {
+        toolbarDisposer: Function;
+
+        componentWillMount() {
+            if (super.componentWillMount) {
```

Comment:
Didn't we say we throw an exception if the `toolbarDisposer`exists in the parent component at this place? And actually it could probably be before defining the `const WithToolbarComponent` here, then there is less runtime overhead :-)

Reply:
correct :) but to be honest :) i have no idea how this works! let us look at this tomorrow.